server/docs/getting-started.md

```diff
@@ -2,106 +2,251 @@
+  public
+  pure
+  returns (bool)
+{
+  return state.stage == Stage.DONE;
+}
+
+function getTurnTaker(AppState memory state)
+  public
+  pure
+  returns (Player)
+{
+  return state.stage == Stage.COMMITTING_NUM ? Player.SECOND : Player.FIRST;
+}
+```
+
+Actions for the game are defined (and constrained) in the `applyAction` function:
+
+```solidity
+function applyAction(AppState memory state, Action memory action)
+  public
+  pure
+  returns (bytes memory)
+{
+  AppState memory nextState = state;
+  if (action.actionType == ActionType.START_GAME) {
+    require(
+      state.stage == Stage.PRE_GAME,
+      "Cannot apply START_GAME on PRE_GAME"
+    );
+    nextState.stage = Stage.COMMITTING_HASH;
+  } else if (action.actionType == ActionType.COMMIT_TO_HASH) {
+    require(
+      state.stage == Stage.COMMITTING_HASH,
+      "Cannot apply COMMIT_TO_HASH on COMMITTING_HASH"
+    );
+    nextState.stage = Stage.COMMITTING_NUM;
+
+    nextState.commitHash = action.actionHash;
+  } else if (action.actionType == ActionType.COMMIT_TO_NUM) {
+    require(
+      state.stage == Stage.COMMITTING_NUM,
+      "Cannot apply COMMITTING_NUM on COMMITTING_NUM"
+    );
+    nextState.stage = Stage.REVEALING;
+
+    nextState.playerSecondNumber = action.number;
+  } else if (action.actionType == ActionType.REVEAL) {
+    require(
+    state.stage == Stage.REVEALING,
+    "Cannot apply REVEALING on REVEALING"
+    );
+    nextState.stage = Stage.DONE;
+
+    nextState.playerFirstNumber = action.playerFirstNumber;
+    nextState.salt = action.salt;
+  } else {
+    revert("Invalid action type");
+  }
+  return abi.encode(nextState);
+}
+```
+
+You should also take a look at `highRoller()` which takes in the deterministic “random” seed made from the submitted player numbers and outputs dice roll totals for each player.
+
+```solidity
+function highRoller(bytes32 randomness)
+    public
+    pure
+    returns(uint8 playerFirstTotal, uint8 playerSecondTotal)
+  {
+    (bytes8 hash1, bytes8 hash2,
+    bytes8 hash3, bytes8 hash4) = cutBytes32(randomness);
+    playerFirstTotal = bytes8toDiceRoll(hash1) + bytes8toDiceRoll(hash2);
+    playerSecondTotal = bytes8toDiceRoll(hash3) + bytes8toDiceRoll(hash4);
+  }
+```
+
+
+----------
 
-1. Instantiate a Counterfactual NodeProvider
-2. Connect the NodeProvider to a blockchain contract through an AppFactory instance
-3. Use the AppFactory’s `.proposeInstallVirtual()` method to propose a new virtual state channel based on the AppFactory instance’s settings (including stakes and game logic)
-4. Use the NodeProvider’s `.on()`  method to listen for accepted installs and updated state in the channel
-5. Use the AppInstance’s `.takeAction()` method to propose updates to state in the channel
-6. Use the AppInstance’s `.uninstall()` method to propose closing and resolving the channel
+# Getting Started
 
 ----------
 
-# HighRoller.js
 
-## Truffle unbox
+## Counterfactual's Truffle box
```

Comment:
This isn't really "Counterfactual's Truffle box". Still a weird header for this section.

Probably "Setting Up The Repo" or something?

Also shouldn't it be way earlier in this guide? We've been referencing these files for some time by this point and only now at this point is the developer / reader of the guide downloading the code.

Probably also reference what a Truffle box is?

"We have packaged all of the code for this getting started guide inside of a Truffle box; a tool for creating templates of Truffle-enabled projects which HighRoller is."